File: S32022.java
import java.io.*;
import java.util.*;
import java.util.ArrayList;
import java.lang.Math;

public class S32022 {
    public static Scanner s = new Scanner(System.in);

    public static void main(String[] args) {
        int n = s.nextInt();
        s.nextLine();
        String str = s.nextLine();
        char[] ch = new char[str.length()];

        for (int i = 0; i < str.length(); i++) ch[i] = str.charAt(i);
        int counter = 0;
        int counter2 = 0;

        for(int i = 1; i < 2 * n; i++){
            if (ch[i] == 'U' && ch[i - 1] == 'U') counter++;
            if (ch[i] == 'I' && ch[i - 1] == 'I') counter2++;
        }

        if (str.charAt(0) == 'U') counter++;

        if (counter2 > counter) System.out.println(counter2);
        else System.out.println(counter);
    }   
}

File: S32022.py
s = int(input())
n = str(input())
counter = 0
counter2 = 0
listOfChars = list()
listOfChars.extend(n)
for i in range(s * 2 - 1):
    if(listOfChars[i] == 'U' and listOfChars[i + 1] == 'U'):
        counter += 1
    if(listOfChars[i] == 'I' and listOfChars[i + 1] == 'I'):
        counter2 += 1
if (listOfChars[0] == 'U'): counter2+=1;
if (counter > counter2):
    print(counter)
else:
    print(counter2)